### ISSUE TITLE
Can't `uv self update` via ssh

### ISSUE THREAD

**Issue by dedebenui** [astral-sh/uv#9250]
I want to manage a remote server with ansible and use `uv`, but the `self update` command doesn't work. I narrowed down the problem to running the command over ssh:
```
ssh remote 'uv --verbose self update'
```
I get the following error:
```
DEBUG uv 0.5.3
DEBUG receipt is not for this executable; assuming uv was installed via a package manager
warning: Self-update is only available for uv binaries installed via the standalone installation scripts.

If you installed uv with pip, brew, or another package manager, update uv with `pip install --upgrade`, `brew upgrade`, or similar.
```

It's the same with `ssh -t`

However, if I first connect to the remote server and then run `uv self update`, it works. Is this a known limitation/bug?

**Comment by zanieb** [comment]
I'm not sure why this is. The function that's used to determine if the receipt is valid is at 

https://github.com/axodotdev/axoupdater/blob/39a6d215c4e4e8f4b8d4c30840019f6b9e0186f7/axoupdater/src/receipt.rs#L81-L109

**Comment by duckinator** [comment]
Hi, sorry you're running into this! I work on the upstream software @zanieb mentioned. 

When you connect to a system via `ssh`, ssh uses a hard-coded PATH by default unless an interactive shell is run. If you do `ssh remote some-command`, you're not running an interactive shell, so it'll use that default PATH.

My suspicion is you have a second copy of `uv`, and `ssh remote uv ...` is running that version. I installed `uv` via the installer, and I get "command not found" instead when I try updating via `ssh localhost uv --verbose self update` (see the screenshot below).

On the system where you had the problem, can you connect normally (in a session where `uv self update` works), then run both `which uv` and `ssh localhost 'which uv'`?

![image](https://github.com/user-attachments/assets/173b4120-fa7b-469f-b29b-e97fd7e6938e)

**Comment by dedebenui** [comment]
Thank you for the detailed reply and the trouble shooting steps. I found that I had some old `uv` binary that I forgot about from when I tried to install it for all user in `/usr/local/bin`. With your explanation on `PATH`, it makes sense that by doing `ssh 'uv self update'`, it ran the binary in `/usr/local/bin` creating conflict with the more recent installation in `~/.local/bin`.

I haven't done more testing, but is that a general problem that can occur when 2 or more different `uv` are installed? Anyway, that was definitely a mistake on my end, thank you again for the help. This specific issue has been solved.
